### DISCUSSION TITLE
Examples

### DISCUSSION THREAD

**Discussion by slyedoc** [emilk/egui#87] [General]
Emilk - First let me say I am new to rust and just learned about egui a week ago, but I am in love.  Immediate mode is an amazing concept I cant believe I didn't know about it and this is new fav repo.

I do have a question about examples, though.  I love rust's practice of the examples folder structure and cargo support.  You have an amazing demo, even go as far as linking to the code from the demo, but that code is actually in egui, not even egui_demo.

Is there any reason not move examples to the rust standard layout.   I think it would clean a few things up and allow easier on boarding. 

This is something I can submit and PR for, but wanted to ask first.  
Its not really an issue, and looked like a decent reason to try out discussions.

Thanks, and sorry if I am totally off base.

**Comment by emilk** [comment]
Hi @slyedoc!

For the last week I have been working on a major refactor of the whole egui folder structure, and I just pushed it to github.

In the new structure, all the example/demo code is now extracted from the `egui` crate to a new crate called `egui_demo_lib`.

The reason I don't have the demo code as cargo examples is because `egui` is not alone enough to *run* the examples - for that one also needs an `egui` backend (such as `egui_web`).

Another reason not to put the example code in a "cargo example" structure is that I plan to publish `egui_demo_lib` to <https://crates.io> so that it can be used by others who want to test their own egui integration with some example code.

The crate `egui_demo_app` uses `egui_demo_lib` + a new crate `eframe` to compile the demo which can be found at <https://emilk.github.io/egui/>. `eframe` has one simple example at <https://github.com/emilk/egui/blob/master/eframe/examples/hello_world.rs>.

All the above crates are still part of <https://github.com/emilk/egui>.
